src/translib/transformer/xfmr_bgp.go

```diff
@@ -0,0 +1,396 @@
+	afName := pathInfo.Var("afi-safi-name")
+	afi := ""
+    var err error
+
+    if len(pathInfo.Vars) < 4 {
+        return afi, errors.New("Invalid Key length")
+    }
+
+    if len(niName) == 0 {
+        return afi, errors.New("vrf name is missing")
+    }
+
+    if strings.Contains(bgpId,"BGP") == false {
+        return afi, errors.New("BGP ID is missing")
+    }
+    
+    if len(protoName) == 0 {
+        return afi, errors.New("Protocol Name is missing")
+    }
+
+	if strings.Contains(afName, "IPV4_UNICAST") {
+		afi = "ipv4_unicast"
+	} else if strings.Contains(afName, "IPV6_UNICAST") {
+		afi = "ipv6_unicast"
+	} else if strings.Contains(afName, "L2VPN_EVPN") {
+		afi = "l2vpn_evpn"
+	} else {
+		log.Info("Unsupported AFI type " + afName)
+        return afi, errors.New("Unsupported AFI type " + afName)
+	}
+
+    if strings.Contains(afName, "IPV4_UNICAST") {
+        afName = "IPV4_UNICAST"
+        if strings.Contains(inParams.uri, "ipv6-unicast") ||
+           strings.Contains(inParams.uri, "l2vpn-evpn") {
+		    err = errors.New("IPV4_UNICAST supported only on ipv4-config container")
+		    log.Info("IPV4_UNICAST supported only on ipv4-config container: ", afName);
+		    return afName, err
+        }
+    } else if strings.Contains(afName, "IPV6_UNICAST") {
+        afName = "IPV6_UNICAST"
+        if strings.Contains(inParams.uri, "ipv4-unicast") ||
+           strings.Contains(inParams.uri, "l2vpn-evpn") {
+		    err = errors.New("IPV6_UNICAST supported only on ipv6-config container")
+		    log.Info("IPV6_UNICAST supported only on ipv6-config container: ", afName);
+		    return afName, err
+        }
+    } else if strings.Contains(afName, "L2VPN_EVPN") {
+        afName = "L2VPN_EVPN"
+        if strings.Contains(inParams.uri, "ipv6-unicast") ||
+           strings.Contains(inParams.uri, "ipv4-unicast") {
+		    err = errors.New("L2VPN_EVPN supported only on l2vpn-evpn container")
+		    log.Info("L2VPN_EVPN supported only on l2vpn-evpn container: ", afName);
+		    return afName, err
+        }
+    } else  {
+	    err = errors.New("Unsupported AFI SAFI")
+	    log.Info("Unsupported AFI SAFI ", afName);
+	    return afName, err
+    }
+
+	key := niName + "|" + afi
+	
+	log.Info("AFI key: ", key)
+
+    return key, nil
+}
+
+var DbToYang_bgp_gbl_afi_safi_key_xfmr KeyXfmrDbToYang = func(inParams XfmrParams) (map[string]interface{}, error) {
+    rmap := make(map[string]interface{})
+    entry_key := inParams.key
+    log.Info("DbToYang_bgp_gbl_afi_safi_key_xfmr: ", entry_key)
+
+    mpathKey := strings.Split(entry_key, "|")
+	afi := ""
+
+	switch mpathKey[1] {
+	case "ipv4_unicast":
+		afi = "IPV4_UNICAST"
+	case "ipv6_unicast":
+		afi = "IPV6_UNICAST"
+	case "l2vpn_evpn":
+		afi = "L2VPN_EVPN"
+	}
+
+    rmap["afi-safi-name"] = afi
+
+	log.Info("DbToYang_bgp_gbl_afi_safi_key_xfmr: rmap:", rmap)
+    return rmap, nil
+}
+
+var DbToYang_bgp_routes_get_xfmr SubTreeXfmrDbToYang = func(inParams XfmrParams) error {
```

Comment:
Everything in this function is commented. Please do note that do not compare the URI for get. You will get the target YGOT structure from which you can compare what is being asked. The advantage with that is we will not have so many string comparisons.

Reply:
Sure Anand, we will take care of that in the next PR. FYI @rathnasabapathyv